p4-16/spec/P4-16-spec.mdk

```diff
@@ -2415,17 +2444,21 @@ infinite-precision integer, without a width specified.
 | `void`         | error     | error   |  error   |
 | `error`        | error     | error   |  allowed |
 | `match_kind`   | error     | error   |  error   |
-| `bool`         | error     | error   |  allowed |
+| `bool`         | allowed   | error   |  allowed |
 | `enum`         | allowed[^enum_header]     | error   |  allowed |
 | `header`       | error     | allowed |  allowed |
 | header stack   | error     | error   |  allowed |
 | `header_union` | error     | error   |  allowed |
-| `struct`       | error     | error   |  allowed |
+| `struct`       | allowed[^struct_header]   | error   |  allowed |
 | `tuple`        | error     | error   |  allowed |
 |----------------|-----------|---------|----------|
 
 [^enum_header]: an `enum` type used as a field in a `header` must specify a
     underlying type and representation for `enum` elements.
+[^struct_header]: a `struct` or nested 'struct' type that has the same properties, 
```

Comment:
Recommend replacing single quotes around the second occurrence of struct with backquotes, so it is like the first occurrence on the same line.